src/main/java/com/cleanroommc/groovyscript/mapper/ItemStackMapper.java

```diff
@@ -0,0 +1,60 @@
+package com.cleanroommc.groovyscript.mapper;
+
+import com.cleanroommc.groovyscript.api.Result;
+import com.cleanroommc.groovyscript.compat.mods.GroovyContainer;
+import com.cleanroommc.groovyscript.server.Completions;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GlStateManager;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.fml.common.registry.ForgeRegistries;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collections;
+import java.util.List;
+
+public class ItemStackMapper extends AbstractObjectMapper<ItemStack> {
+
+    public static final ItemStackMapper INSTANCE = new ItemStackMapper("item", null);
+
+    protected ItemStackMapper(String name, GroovyContainer<?> mod) {
+        super(name, mod, ItemStack.class);
+        addSignature(String.class, int.class);
+        this.documentation = docOfType("item stack");
+    }
+
+    @Override
+    public Result<ItemStack> getDefaultValue() {
+        return Result.some(ItemStack.EMPTY);
+    }
+
+    @Override
+    public @NotNull Result<ItemStack> parse(String mainArg, Object[] args) {
+        return ObjectMappers.parseItemStack(mainArg, args);
+    }
+
+    @Override
+    public void provideCompletion(int index, Completions items) {
+        if (index == 0) items.addAllOfRegistry(ForgeRegistries.ITEMS);
```

Comment:
is it possible for the valid metadata of items to have autocompletion? pretty sure its a "no"

Reply:
It is possible, but id argue that its not worth it